apps/src/aichat/views/ChatWorkspace.tsx

```diff
@@ -133,14 +136,59 @@ const ChatWorkspace: React.FunctionComponent<ChatWorkspaceProps> = ({
     tabPanelsContainerClassName: moduleStyles.tabPanelsContainer,
   };
 
-  const onCloseWarningModal = useCallback(
-    () => dispatch(setShowWarningModal(false)),
-    [dispatch]
-  );
+  const onCloseModal = useCallback(() => {
+    if (
+      isUserTeacher &&
+      tryGetLocalStorage('teacherSawOnboarding', 'no') !== 'yes'
+    ) {
+      trySetLocalStorage('teacherSawOnboarding', 'yes');
+    }
+    dispatch(setShowWarningModal(false));
+    dispatch(setShowOnboardingModal(false));
+  }, [dispatch, isUserTeacher]);
+
+  const isTeacherFirstAichatEncounter = useCallback(() => {
+    if (isUserTeacher) {
+      const teacherSawOnboarding = tryGetLocalStorage(
+        'teacherSawOnboarding',
+        'no'
+      );
+      if (teacherSawOnboarding !== 'yes') {
+        if (showOnboardingModal) {
+          // The first time a teacher opens an aichat level, the onboarding modal is displayed.
+          // The onboarding modal remains displayed until the teacher closes the modal.
+          // When the teacher closes the onboarding modal, the 'teacherSawOnboarding' value
+          // is set to 'yes'.
+          trySetLocalStorage('teacherSawOnboarding', 'inProgress');
```

Comment:
What does the `'inProgress'` value mean here? Could we just leave it unset and get the same effect?

Reply:
Oh yeah - can leave unset. Updated!